padd.sh

```diff
@@ -1370,6 +1370,40 @@ NormalPADD() {
   done
 }
 
+Update() {
+    GetPADDInformation
+
+    if [ "${padd_out_of_date_flag}" = "true" ]; then
+        echo "${full_status_update}"
+
+        # Get script path
+        padd_script_path="${BASH_SOURCE}"
```

Comment:
Where is this variable declared (`BASH_SOURCE`)?

Reply:
It's a predefined variable:

> BASH_SOURCE
>
> An array variable whose members are the source filenames where the corresponding shell function names in the FUNCNAME array variable are defined. The shell function ${FUNCNAME[$i]} is defined in the file ${BASH_SOURCE[$i]} and called from ${BASH_SOURCE[$i+1]}

See https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#Bash-Variables

On a side-note this should be `${BASH_SOURCE:-$0}` according to https://www.appsloveworld.com/bash/100/2/choosing-between-0-and-bash-source since the stickler-ci checks suggests `padd.sh#L1380 Expanding an array without an index only gives the first element.`